signature/cose/envelope.go

```diff
@@ -486,6 +493,28 @@ func generateUnprotectedHeaders(req *signature.SignRequest, signer signer, unpro
 		certChain[i] = c.Raw
 	}
 	unprotected[cose.HeaderLabelX5Chain] = certChain
+
+	// timestamping
+	if signingScheme == string(signature.SigningSchemeX509) && req.TSAServerURL != "" {
+		hash := hashFunc(signer.Algorithm())
+		nonce, err := timestamp.GenerateNonce(rand.Reader)
+		if err != nil {
+			return &signature.TimestampError{Detail: err}
+		}
+		timeStampOpts := tspclient.RequestOptions{
+			Content:                 sig,
+			HashAlgorithm:           hash,
+			HashAlgorithmParameters: asn1.NullRawValue,
```

Comment:
The zero value for https://pkg.go.dev/encoding/asn1#RawValue has `Tag = 0`, whereas https://pkg.go.dev/encoding/asn1#NullRawValue has `Tag = 5`. Otherwise, they should be the same.

Reply:
Created PR in tspclient-go as a fix: https://github.com/notaryproject/tspclient-go/pull/24